doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+
+# Start the default profile in directory "c:/Users/Foo/dev/MyProject" (user story 8)
+wt new-tab --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -d "c:/Users/Foo/dev/MyProject"
+
+# Runs the user's "Windows Powershell" profile in a new tab in directory
+#  "c:/Users/Foo/dev/MyProject" (user story 2, 8)
+wt new-tab --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -p "Windows Powershell" -d "c:/Users/Foo/dev/MyProject"
+
+# open a new tab with the "Windows Powershell" profile, and another with the
+#  "cmd" profile (user story 12)
+wt new-tab --profile "Windows Powershell" ; new-tab --profile "cmd"
+wt --profile "Windows Powershell" ; new-tab --profile "cmd"
+wt --profile "Windows Powershell" ; --profile "cmd"
+wt --p "Windows Powershell" ; --p "cmd"
+
+# run "my-commandline.exe with some args" in a new tab
+wt new-tab my-commandline.exe with some args
+wt my-commandline.exe with some args
+
+# run "my-commandline.exe with some args and a ; literal semicolon" in a new
+#  tab, and in another tab, run "another.exe running in a second tab"
+wt my-commandline.exe with some args and a \; literal semicolon ; new-tab another.exe running in a second tab
+
+# Start cmd.exe, then split it vertically (with the first taking 70% of it's
+#  space, and the new pane taking 30%), and run wsl.exe in that pane (user story 13)
+wt cmd.exe ; split-pane -t 0 -v -P 30 wsl.exe
+wt cmd.exe ; split-pane -P 30 wsl.exe
+
+# Create a new window with the default profile, create a vertical split with the
+#  default profile, then create a horizontal split in the second pane and run
+#  "media.exe" (user story 13)
+wt new-tab ; split-pane -v ; split-pane -t 1 -h media.exe
+
+```
+
+## `wt` Syntax
+
+The `wt` commandline is divided into two main sections: "Options", and "Commands":
+
+`wt [options] [command ; ]...`
+
+Options are a list of flags and other parameters that can control the behavior
+of the `wt` commandline as a whole. Commands are a semicolon-delimited list of
+commands and arguments for those commands.
+
+If no command is specified in a `command`, then the command is assumed to be a
+`new-tab` command by default. So, for example, `wt cmd.exe` is interpreted the
+same as `wt new-tab cmd.exe`.
+
+<!--
+### Aside: What should the default command be?
+
+These are notes from my draft intentionally left here to help understand the
+conclusion that new-tab should be the default command.
+
+Should the default command be `new-window` or `new-tab`?
+
+`new-window` makes sense to take params like `--initialPosition`,
+`--initialRows`/`--initialCols`, and _implies_ `new-tab`. However, chained
+commands that want to open in the same window _need_ to specify `new-tab`,
+otherwise they'll all appear in new windows.
+
+If it's `new-tab`, then how do `--initialRows` (etc) work? `new-tab` generally

```

Comment:
I prefer the idea of `new-tab` being the default.

Why not treat `new-window` as a flag, rather than a command? Then, if the flag is present, we can also support the other ones mentioned like `--initialRows`, `--initialCols`, etc...